clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -326,16 +327,15 @@ public static void main(String[] args) {
 
     private static boolean retainsBufferReference(Schema schema) {
         final AtomicReference<Boolean> foundBufferReference = new AtomicReference<>(Boolean.FALSE);
-        SchemaVisitor detector = new SchemaVisitorAdapter() {
+        SchemaVisitor detector = new SchemaVisitor() {
             @Override
             public void visit(Type field) {
-                if (field == BYTES || field == NULLABLE_BYTES || field == RECORDS) {
+                if (field == BYTES || field == NULLABLE_BYTES || field == RECORDS)
                     foundBufferReference.set(Boolean.TRUE);
-                }
             }
         };
         foundBufferReference.set(Boolean.FALSE);
```

Comment:
This seems redundant.

Reply:
Yeah, simplified it all in my latest commit.